### PR TITLE
[REF] mass_mailing: onchange to computed fields

### PR DESCRIPTION
**Pull request by ryv-odoo**
onchange acts only in views, to avoid future issue about that
we cover the onchange behaviour with computed storable and not
readonly fields.

task-2088577

### CODE REVIEW

File: addons/mass_mailing/models/mailing.py [hunk 1]
```diff
@@ -197,8 +198,8 @@ def _compute_next_departure(self):
             else:
                 mass_mailing.next_departure = cron_time
 
-    @api.onchange('mailing_model_name', 'contact_list_ids')
-    def _onchange_model_and_list(self):
+    @api.depends('mailing_model_name', 'contact_list_ids')
+    def _compute_mailing_domain(self):
         mailing_domain = literal_eval(self.mailing_domain) if self.mailing_domain else []
```
Comment (dbeguin) [inline addons/mass_mailing/models/mailing.py]: for mailing in self

File: addons/mass_mailing/models/mailing.py [hunk 2]
```diff
@@ -217,8 +218,8 @@ def _onchange_model_and_list(self):
             mailing_domain = []
         self.mailing_domain = repr(mailing_domain)
 
-    @api.onchange('mailing_type')
-    def _onchange_mailing_type(self):
+    @api.depends('mailing_type')
+    def _compute_medium_id(self):
         if self.mailing_type == 'mail' and not self.medium_id:
```
Comment (dbeguin) [inline addons/mass_mailing/models/mailing.py]: same here

File: addons/mass_mailing_event/models/mailing_mailing.py [hunk 3]
```diff
@@ -4,10 +4,11 @@
...
-    @api.onchange('mailing_model_real', 'contact_list_ids')
-    def _onchange_model_and_list(self):
+    @api.depends('mailing_model_real', 'contact_list_ids')
+    def _compute_mailing_domain(self):
         # TDE FIXME: whuuut ?
-        result = super(MassMailingCampaign, self)._onchange_model_and_list()
-        if self.mailing_model_name == 'event.registration' and self.mailing_domain == '[]':
-            self.mailing_domain = self.env.context.get('default_mailing_domain', '[]')
+        result = super(MassMailingCampaign, self)._compute_mailing_domain()
+        for mailing in self:
```
Comment (dbeguin) [inline addons/mass_mailing_event/models/mailing_mailing.py]: and you did it here ! Nice ! :D

**Comment by tde-banana-odoo** [comment]
Overseeded by https://github.com/odoo/odoo/pull/41877 .

**Pull request closed without merging**